libvirt/image/Makefile

```diff
@@ -120,9 +125,32 @@ $(PAUSE_SRC): $(SKOPEO_SRC)/bin/skopeo
 $(PAUSE): | $(PAUSE_SRC) $(UMOCI_SRC)/umoci
 	$(UMOCI_SRC)/umoci unpack --rootless --image "$(PAUSE_SRC):$(PAUSE_VERSION)" "${FILES_DIR}/$(PAUSE_BUNDLE)"
 
+docker-build:
+	docker build -t caa-libvirt-ubuntu-podvm-builder \
+	-f Dockerfile_ubuntu.podvm_builder .
+	docker run --rm \
+		-v $(PROJECT_DIR):/src/cloud-api-adaptor \
+		-v $(abspath $(PROJECT_DIR)/../kata-containers):/src/kata-containers \
+		-e UID=$(shell id -u) \
+		-e GID=$(shell id -g) \
+		caa-libvirt-ubuntu-podvm-builder \
+		bash -c 'trap "chown -R -f "$$UID:$$GID" output staticlib $(SKOPEO_SRC) \
+		$(UMOCI_SRC) $(PAUSE_SRC) $(FILES_DIR)/$(PAUSE_BUNDLE) $(FILES_DIR)/usr \
+		../../agent-protocol-forwarder || true" EXIT; \
+		source ~/.cargo/env && CLOUD_PROVIDER=libvirt make build'
```

Comment:
I also hit a packer error here:
```
packer build -var qemu_image_name=podvm-4e38611-dirty-amd64.qcow2 \
	-var cloud_image_url=https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img \
	-var cloud_image_checksum=b6d04d7ed478a083547eebf009228dc72afade79fcbc097f95226512056bbc2e .
qemu.ubuntu: output will be in this color.

==> qemu.ubuntu: Retrieving ISO
==> qemu.ubuntu: Trying https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img
==> qemu.ubuntu: Trying https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img?checksum=sha256%3Ab6d04d7ed478a083547eebf009228dc72afade79fcbc097f95226512056bbc2e
==> qemu.ubuntu: https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img?checksum=sha256%3Ab6d04d7ed478a083547eebf009228dc72afade79fcbc097f95226512056bbc2e => /root/.cache/packer/36780b83a751001e19d6763c24d4845672af5f33.iso
==> qemu.ubuntu: Copying hard drive...
==> qemu.ubuntu: Resizing hard drive...
==> qemu.ubuntu: Found port for communicator (SSH, WinRM, etc): 2257.
==> qemu.ubuntu: Looking for available port between 5900 and 6000 on 127.0.0.1
==> qemu.ubuntu: Starting VM, booting disk image
    qemu.ubuntu: The VM will be run headless, without a GUI. If you want to
    qemu.ubuntu: view the screen of the VM, connect via VNC without a password to
    qemu.ubuntu: vnc://127.0.0.1:5965
==> qemu.ubuntu: Overriding default Qemu arguments with qemuargs template option...
==> qemu.ubuntu: Waiting 10s for boot...
==> qemu.ubuntu: Connecting to VM via VNC (127.0.0.1:5965)
==> qemu.ubuntu: Typing the boot command over VNC...
    qemu.ubuntu: Not using a NetBridge -- skipping StepWaitGuestAddress
==> qemu.ubuntu: Using SSH communicator to connect: 127.0.0.1
==> qemu.ubuntu: Waiting for SSH to become available...
==> qemu.ubuntu: Error waiting for SSH: Packer experienced an authentication error when trying to connect via SSH. This can happen if your username/password are wrong. You may want to double-check your credentials as part of your debugging process. original error: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none], no supported methods remain
==> qemu.ubuntu: Deleting output directory...
Build 'qemu.ubuntu' errored after 4 minutes 50 seconds: Packer experienced an authentication error when trying to connect via SSH. This can happen if your username/password are wrong. You may want to double-check your credentials as part of your debugging process. original error: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none], no supported methods remain

==> Wait completed after 4 minutes 50 seconds

==> Some builds didn't complete successfully and had errors:
--> qemu.ubuntu: Packer experienced an authentication error when trying to connect via SSH. This can happen if your username/password are wrong. You may want to double-check your credentials as part of your debugging process. original error: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none], no supported methods remain

==> Builds finished but no artifacts were created.
make: *** [Makefile:84: podvm-4e38611-dirty-amd64.qcow2] Error 1
make: *** [Makefile:130: docker-build] Error 2
```
It might be user error, but as I think it is inside the container I'm not sure how to debug it?

Reply:
Packer start qemu inside the container. It seems not able to connect to the VM via ssh after a while. In `qemu-ubuntu.pkr.hcl` it set `ssh_wait_timeout` to `300s`. Could you increase the timeout and give it a try?

Reply:
So I tried to double the timeout to 10mins:
```
diff --git a/libvirt/image/qemu-ubuntu.pkr.hcl b/libvirt/image/qemu-ubuntu.pkr.hcl
index 083d6ad..12e1564 100644
--- a/libvirt/image/qemu-ubuntu.pkr.hcl
+++ b/libvirt/image/qemu-ubuntu.pkr.hcl
@@ -12,7 +12,7 @@ source "qemu" "ubuntu" {
   ssh_password      = "${var.ssh_password}"
   ssh_port          = 22
   ssh_username      = "${var.ssh_username}"
-  ssh_wait_timeout  = "300s"
+  ssh_wait_timeout  = "600s"
   vm_name           = "${var.qemu_image_name}"
   shutdown_command  = "sudo shutdown -h now"
 }
```

But it still failed just short of 5 again:
```
==> qemu.ubuntu: Waiting for SSH to become available...
==> qemu.ubuntu: Error waiting for SSH: Packer experienced an authentication error when trying to connect via SSH. This can happen if your username/password are wrong. You may want to double-check your credentials as part of your debugging process. original error: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none], no supported methods remain
==> qemu.ubuntu: Deleting output directory...
Build 'qemu.ubuntu' errored after 4 minutes 57 seconds: Packer experienced an authentication error when trying to connect via SSH. This can happen if your username/password are wrong. You may want to double-check your credentials as part of your debugging process. original error: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none], no supported methods remain
```

So I don't know if that was enough. Sorry for being slow here - I'm exercising the path of a user who has no clue what they are doing 😉